ow_faults_detection/include/ow_faults_detection/FaultDetector.h

```diff
@@ -57,24 +57,20 @@ class FaultDetector
   // Get index from m_joint_index_map. If found, modify out_index and return
   // true. Otherwise, return false.
   bool findJointIndex(const unsigned int joint, unsigned int& out_index);
-
-  // Antenna
-  void antPublishFaultMessages();
   
   // Camera
   void cameraTriggerCb(const std_msgs::Empty& msg);
   void cameraRawCb(const sensor_msgs::Image& msg);
-  void cameraPublishFaultMessages(bool is_fault);
 
   // Power
-  void publishPowerSystemFault();
   void powerSOCListener(const owl_msgs::BatteryStateOfCharge& msg);
   void powerTemperatureListener(const owl_msgs::BatteryTemperature& msg);
 
   // OWLAT MESSAGE FUNCTIONS AND PUBLISHERS
-  void publishSystemFaultsMessage();
   template<typename fault_msg>
   void setFaultsMessageHeader(fault_msg& msg);
+  template<typename faults_msg_pub_t, typename faults_msg_t, typename faults_flags_t>
+  void publishFaultsMessage(faults_msg_pub_t& faults_msg_pub, faults_msg_t& faults_msg, faults_flags_t faults_flags);
```

Comment:
Just  readability suggestion: parameter names can be omitted in declarations, and are not helpful here since the typename tells all.  This would also shorten the line.